upgrade/scripts/upgrade/rollout-restart.sh

```diff
@@ -0,0 +1,259 @@
+#!/bin/bash
+
+# MIT License
+#
+# (C) Copyright 2024 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# When upgrading from Istio 1.11.8 to 1.19.10 we need to rollout-restart because
+# the istio-injection enabled namespaces doesn't get the latest image of istio.
+#
+
+set -x
+
+# This part of script is to delete the pods on vault namespace which do not have proxyv2:1.19.10
+# This is done because the vault is not updated in CSM1.6, this restart of vault may not be needed in future
+
+NAMESPACE="vault"
+CORRECT_VERSION="proxyv2:1.19.10"
+pods_deleted=false
+
+# Get all pods in the namespace
+pods=$(kubectl get pods -n "$NAMESPACE" -o jsonpath='{.items[*].metadata.name}')
+
+# Loop through each pod
+for pod in $pods; do
+  echo "Checking pod: $pod"
+
+  # Get the image version of proxyv2 container in the pod
+  image=$(kubectl get pod "$pod" -n $NAMESPACE -o jsonpath='{.spec.containers[*].image}')
+
+  # If the image version is not correct, delete the pod
+  if [[ $image != *"$CORRECT_VERSION"* ]]; then
+    echo "Deleting pod: $pod (Incorrect proxyv2 version: $image)"
+    kubectl delete pod "$pod" -n "$NAMESPACE" --grace-period=0 --force
+    pods_deleted=true
+  else
+    echo "Pod $pod has the correct proxyv2 version: $image"
+  fi
+done
+
+# Wait for 2 minutes if any pods were deleted
+if [ "$pods_deleted" = true ]; then
+  echo "Waiting for 2 minutes for pods to restart..."
+  sleep 120
+
+  # Check the status of the pods after restart
+  new_pods=$(kubectl get pods -n "$NAMESPACE")
+
+  echo "Checking the status of new pods..."
+  echo "$new_pods"
+
+  # Check if all pods are in Running state and have the correct proxyv2 version
+  all_correct=true
+  for pod in $(echo "$new_pods" | awk '{print $1}' | grep -v NAME); do
```

Comment:
This can be simply
```
for pod in $new_pods; do
```
if you change the fetching of the pods on line 64 to:
```
new_pods=$(kubectl get pods -n "$NAMESPACE" -o jsonpath='{.items[*].metadata.name}')
```
just like the query on line 39.